# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/gamepad/GamepadMonitoring.cpp
# Commit: b9e6371846e0
# Full Hash: b9e6371846e0ae62c4cd5826c182f7b12eec1620
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-08-07 21:36:18
# Regressor Bug: 1657404
# File Overlap Count: 1
# Description:
#   Bug 1657404 - Refactor gamepad monitoring - Part 2 r=daoshengmu
#   
#   Substitute and eliminate MaybeStopGamepadMonitoring(). The logic to do this
#   is so dependent on GamepadPlatformService that it should probably just be a
#   member.
# ==============================================================================

diff -r 550630983bc5 -r b9e6371846e0 dom/gamepad/GamepadMonitoring.cpp
--- a/dom/gamepad/GamepadMonitoring.cpp	Fri Aug 07 13:32:17 2020 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,30 +0,0 @@
-/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set ts=8 sts=2 et sw=2 tw=80: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#include "mozilla/dom/GamepadMonitoring.h"
-#include "mozilla/dom/GamepadPlatformService.h"
-#include "mozilla/ipc/BackgroundParent.h"
-
-using namespace mozilla::ipc;
-
-namespace mozilla {
-namespace dom {
-
-void MaybeStopGamepadMonitoring() {
-  AssertIsOnBackgroundThread();
-  RefPtr<GamepadPlatformService> service =
-      GamepadPlatformService::GetParentService();
-  MOZ_ASSERT(service);
-  if (service->HasGamepadListeners()) {
-    return;
-  }
-  StopGamepadMonitoring();
-  service->ResetGamepadIndexes();
-  service->MaybeShutdown();
-}
-
-}  // namespace dom
-}  // namespace mozilla